cmd/osmosisd/cmd/root.go

```diff
@@ -85,6 +89,28 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 	return rootCmd, encodingConfig
 }
 
+func getHomeEnvironment() string {
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		panic(err)
+	}
+	envPath := filepath.Join(userHomeDir, ".osmosisd/.env")
+
+	err = godotenv.Load(envPath)
+	if err != nil {
+		panic(err)
+	}
+
+	val := os.Getenv(EnvVariable)
+	if val == EnvMainnet {
+		return filepath.Join(userHomeDir, ".osmosisd")
+	} else if val == EnvLocalnet {
+		return filepath.Join(userHomeDir, ".osmosisd-local")
+	} else {
+		panic(fmt.Errorf("Invalid environment variable"))
```

Comment:
Wait the env val can be anything, we shouldn't restrict it. These are just the two common ones, no?

Like you can have a different filepath.

Also we need to ensure commands work with a different `--home` flag, and nothing in `.osmosisd`